assets/js/cell/index.js

```diff
@@ -43,110 +63,102 @@ const Cell = {
         });
       }
 
-      // Setup action handlers
-      if (this.props.type === "elixir") {
-        const amplifyButton = this.el.querySelector(
-          `[data-element="amplify-outputs-button"]`
-        );
-        amplifyButton.addEventListener("click", (event) => {
-          this.el.toggleAttribute("data-js-amplified");
-        });
-      }
+      const editorContainer = this.el.querySelector(
+        `[data-element="editor-container"]`
+      );
+      // Remove the content placeholder.
+      editorContainer.firstElementChild.remove();
+      // Create an empty container for the editor to be mounted in.
+      const editorElement = document.createElement("div");
+      editorContainer.appendChild(editorElement);
+      // Setup the editor instance.
+      const language = {
+        markdown: "markdown",
+        elixir: "elixir",
+        smart: "elixir",
+      }[this.props.type];
+      const readOnly = this.props.type === "smart";
+      this.state.liveEditor = new LiveEditor(
+        this,
+        editorElement,
+        this.props.cellId,
+        source,
+        revision,
+        language,
+        readOnly
+      );
+
+      // Setup change indicator
+      if (isEvaluable(this.props.type)) {
+        this.state.evaluationDigest = evaluation_digest;
+
+        const updateChangeIndicator = () => {
+          const cellStatus = this.el.querySelector(
+            `[data-element="cell-status"]`
+          );
+          const indicator =
+            cellStatus &&
+            cellStatus.querySelector(`[data-element="change-indicator"]`);
+
+          if (indicator) {
+            const source = this.state.liveEditor.getSource();
+            const digest = md5Base64(source);
+            const changed = this.state.evaluationDigest !== digest;
+            cellStatus.toggleAttribute("data-js-changed", changed);
+          }
+        };
 
-      // Setting up an editor takes relatively much synchronous time,
-      // so we postpone it, so that other immediate initializations
-      // are done first. This is relevant if there are a lot of cells
-      setTimeout(() => {
```

Comment:
I needed to kill this optimisation for now, because the smart cell sends delta right away and so we want to setup all event handlers immediately.